main.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"runtime"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	"github.com/operator-framework/operator-lib/leader"
+	_ "k8s.io/client-go/plugin/pkg/client/auth"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorconfig "github.com/openshift/configure-alertmanager-operator/config"
+	operatormetrics "github.com/openshift/configure-alertmanager-operator/pkg/metrics"
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	managedopenshiftiov1alpha1 "github.com/openshift/configure-alertmanager-operator/api/v1alpha1"
+	"github.com/openshift/configure-alertmanager-operator/controllers"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme            = k8sruntime.NewScheme()
+	setupLog          = ctrl.Log.WithName("setup")
+	metricsHost       = "0.0.0.0"
+	metricsPort int32 = 8383
+)
+
+var log = logf.Log.WithName("cmd")
+
+func printVersion() {
+	log.Info(fmt.Sprintf("Go Version: %s", runtime.Version()))
+	log.Info(fmt.Sprintf("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH))
+}
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(configv1.AddToScheme(scheme))
+	utilruntime.Must(monitoringv1.AddToScheme(scheme))
+
+	utilruntime.Must(managedopenshiftiov1alpha1.AddToScheme(scheme))
```

Comment:
Is this needed? As there is nothing to add here?

Reply:
Given there is no API at all, we can perhaps explore removing the contents of the entire `api` directory entirely (but put a `.gitkeep` in there similar to what MNMO has done, so that the directory is retained to prevent breakage of OSDK or boilerplate), and it can be removed from here at the same time.

Reply:
I am thinking almost the same, if we do not need to add scheme here, so maybe the things under `apis/` can be removed as well. Just not sure how would it impact the operator.

But it is a bit weird that the operator does not have new api resource defined.

Reply:
I was wondering what to do about this too.
Thanks for the suggestions I will make those changes and see how it goes.